src/coreclr/jit/codegenarm.cpp

```diff
@@ -1360,7 +1360,7 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
             instGen_MemoryBarrier();
         }
 
-        GetEmitter()->emitInsLoadStoreOp(ins_Store(type), emitActualTypeSize(type), data->GetRegNum(), tree);
+        GetEmitter()->emitInsLoadStoreOp(ins_Store(data->TypeGet()), emitActualTypeSize(type), data->GetRegNum(), tree);
```

Comment:
It looks like the assert at the top didn't help us here:
```
assert(!varTypeIsFloating(type) || (type == data->TypeGet()));
```

Should that be updated?

Also, should you just set `type = data->TypeGet()` instead, so the `emitActualTypeSize` also gets the "new" type?

Reply:
That's true...`emitAttr` interpretation is little confusing (we already have https://github.com/dotnet/runtime/issues/12178 to improve it). I will fix this.